include/boost/geometry/formulas/andoyer_inverse.hpp

```diff
@@ -131,6 +150,11 @@ class andoyer_inverse
                 // azimuth = -inf
                 result.azimuth = lat1 <= lat2 ? c0 : pi;
             }
+            // meridian segment
+            else if (math::equals(lon1, lon2))
+            {
+                result.azimuth = c0;
```

Comment:
The azimuth can also be `pi` in this case if `lat2 < lat1`. Maybe also -pi if `lon2 < lon1`, so very close to each other, idk.